mypy/checker.py

```diff
@@ -1676,6 +1677,60 @@ def visit_class_def(self, defn: ClassDef) -> None:
         if typ.is_protocol and typ.defn.type_vars:
             self.check_protocol_variance(defn)
 
+    def check_init_subclass(self, defn: ClassDef) -> None:
+        """
+        Check that keywords in a class definition are valid arguments for __init_subclass__().
+
+        In this example:
+            1   class Base:
+            2       def __init_subclass__(cls, thing: int):
+            3           pass
+            4   class Child(Base, thing=5):
+            5       def __init_subclass__(cls):
+            6           pass
+            7   Child()
+
+        Base.__init_subclass__(thing=5) is called at line 4. This is what we simulate here.
+        Child.__init_subclass__ is never called.
+        """
+        typ = defn.info
+        # At runtime, only Base.__init_subclass__ will be called
+        # we skip the current class itself.
+        found = False
+        for base in typ.mro[1:]:
+            # there are "NOT_READY" instances
+            # during the tests, so I filter them out...
+            if base.defn.info:
+                for method_name, method_symbol_node in base.defn.info.names.items():
```

Comment:
You don't need this to be a cycle, just use `method_symbol_node = base.names.get('__init_subclass__')`.

Reply:
Or even simpler (if you are not going to use `method_symbol_node`):
```python
if '__init_subclass__' not in base.names:
    continue
```

Reply:
you are right, I have modified accordingly